src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -521,19 +526,119 @@ private void VisitConditionalBranch(IOperation condition, ref BasicBlock dest, b
                 default:
                     condition = Operation.SetParentOperation(Visit(condition), null);
                     dest = dest ?? new BasicBlock(BasicBlockKind.Block);
-                    LinkBlocks(CurrentBasicBlock, (condition, sense, dest));
+                    LinkBlocks(CurrentBasicBlock, (condition, sense ? ConditionalBranchKind.IfTrue : ConditionalBranchKind.IfFalse, dest));
                     _currentBasicBlock = null;
                     return;
             }
         }
 
-        private static void LinkBlocks(BasicBlock previous, (IOperation Condition, bool JumpIfTrue, BasicBlock Destination) next)
+        private static void LinkBlocks(BasicBlock previous, (IOperation Value, ConditionalBranchKind Kind, BasicBlock Destination) next)
         {
-            Debug.Assert(previous.Conditional.Condition == null);
+            Debug.Assert(previous.Conditional.Value == null);
+            Debug.Assert(next.Value != null);
+            Debug.Assert(next.Value.Parent == null);
             next.Destination.AddPredecessor(previous);
             previous.Conditional = next;
         }
 
+        public override IOperation VisitCoalesce(ICoalesceOperation operation, int? captureIdForResult)
+        {
+            SyntaxNode valueSyntax = operation.Value.Syntax;
+            ITypeSymbol valueTypeOpt = operation.Value.Type;
+
+            SpillEvalStack();
+
+            int save_availableCaptureId = _availableCaptureId;
```

Comment:
Perhaps `previousAvailableCaptureId` or `saveAvailableCaptureId`, to match naming convention.